src/parse/grammar/checks.ts

```diff
@@ -427,6 +430,53 @@ export function validateAmbiguousAlternationAlternatives(topLevelRule:gast.Rule,
     return errors
 }
 
+export class RepitionCollector extends gast.GAstVisitor {
+    public allProductions:gast.IProduction[] = []
+
+    public visitRepetitionWithSeparator(manySep:gast.RepetitionWithSeparator):void {
+        this.allProductions.push(manySep)
+    }
+
+    public visitRepetitionMandatory(atLeastOne:gast.RepetitionMandatory):void {
+        this.allProductions.push(atLeastOne)
+    }
+
+    public visitRepetitionMandatoryWithSeparator(atLeastOneSep:gast.RepetitionMandatoryWithSeparator):void {
+        this.allProductions.push(atLeastOneSep)
+    }
+
+    public visitRepetition(many:gast.Repetition):void {
+        this.allProductions.push(many)
+    }
+}
+
+export function validateSomeNonEmptyLookaheadPath(topLevelRules:gast.Rule[], maxLookahead:number):IParserDefinitionError[] {
+    let errors = []
+    forEach(topLevelRules, (currTopRule) => {
+        let collectorVisitor = new RepitionCollector()
+        currTopRule.accept(collectorVisitor)
+        let allRuleProductions = collectorVisitor.allProductions
+        forEach(allRuleProductions, (currProd) => {
+            let prodType = getProdType(currProd)
+            let currOccurrence = currProd.occurrenceInParent
+            let paths = getLookaheadPathsForOptionalProd(currOccurrence, currTopRule, prodType, maxLookahead)
+            let pathsInsideProduction = paths[0]
+            if (isEmpty(flatten(pathsInsideProduction))) {
+                let dslName = getProductionDslName(currProd)
+                let errMsg = `Empty lookahead paths found in <${dslName}${currOccurrence}> within Rule <${currTopRule.name}>.\n` +
```

Comment:
The error message is much better now.
But it still needs some work.

'empty lookahead paths' is an internal implementation detail. 
Not quite sure how to describe this yet in a good way for end users.

One option.
> The repetition <MANY3> within Rule <QualifiedName> can never consume any tokens.
    This could lead to an infinite loop. 

Although this pull request does not have to have the "final" version of the message because it will have to change anyhow...

Because once the issue of avoiding (at runtime) entering the repetition for empty lookahead.
Than instead of of this causing infinite loops it would just mean a part of the grammar is redundant.